### DISCUSSION TITLE
How to build the project in Mac with M1 chip?

### DISCUSSION THREAD

**Discussion by zhexuany** [apache/apisix#8369] [Q&A] [answered]
# What have I did?
1. clone the project
2. execute the command I found online to init the environment  `curl https://raw.githubusercontent.com/apache/apisix/master/utils/install-dependencies.sh -sL | bash`
3. make init
4. make dpes

# What is  the error reported?

```
Installing https://luarocks.org/jsonschema-0.9.8-0.src.rock
Missing dependencies for jsonschema 0.9.8-0:
   lrexlib-pcre 2.9.1-1 (not installed)

jsonschema 0.9.8-0 depends on net-url (1.1-1 installed)
jsonschema 0.9.8-0 depends on lrexlib-pcre 2.9.1-1 (not installed)
Installing https://luarocks.org/lrexlib-pcre-2.9.1-1.src.rock

Error: Failed installing dependency: https://luarocks.org/jsonschema-0.9.8-0.src.rock - Failed installing dependency: https://luarocks.org/lrexlib-pcre-2.9.1-1.src.rock - Could not find header file for PCRE
  No file pcre.h in /usr/local/include
  No file pcre.h in /usr/include
  No file pcre.h in /include
  No file pcre.h in /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include
  No file pcre.h in /usr/local/opt/include
```

From the error message, the problem still exists even I already tried to install `jsonschema` manually by providing the correct `PATH` info(well, I installed PCRE via homebrew).

**Comment by zhexuany** [comment]
@soulbird  reminds me to take a look over the [place](https://github.com/apache/apisix/issues/4945#issuecomment-909850338) having a lot discussion about `pcre`. 

It seems to be the answer.

**Comment by zhexuany** [comment, marked as answer]
```
variables = {
   OPENSSL_INCDIR = "/opt/homebrew/opt/openresty-openssl111/include",
   OPENSSL_LIBDIR = "/opt/homebrew/opt/openresty-openssl111/lib",

   PCRE_INCDIR = "/opt/homebrew/include"
}
```

Here is what I did in `luarocks` config file.

**Reply by tokers** [reply]
It's worth writing it to the FAQ :)
